docs/source/general/data.rst

```diff
@@ -2,11 +2,429 @@
 Data
 ####
 
+.. _data:
+
+
+***********
+Terminology
+***********
+
+Here are common terms you need to be familiar with:
+
+.. list-table:: Terminology
+   :widths: 20 80
+   :header-rows: 1
+
+   * - Term
+     - Definition
+   * - :class:`~flash.data.data_module.DataModule`
+     - The :class:`~flash.data.data_module.DataModule` contains the dataset, transforms and dataloaders.
+   * - :class:`~flash.data.data_pipeline.DataPipeline`
+     - The :class:`~flash.data.data_pipeline.DataPipeline` is Flash internal object to manage :class:`~flash.data.process.Preprocess` and :class:`~flash.data.process.Postprocess` objects.
+   * - :class:`~flash.data.process.Preprocess`
+     - The :class:`~flash.data.process.Preprocess` provides a simple hook-based API to encapsulate your pre-processing logic.
+        The :class:`~flash.data.process.Preprocess` provides multiple hooks such as :meth:`~flash.data.process.Preprocess.load_data`
+        and :meth:`~flash.data.process.Preprocess.load_sample` which are used to replace a traditional `Dataset` logic.
+        Flash DataPipeline contains a system to call the right hooks when needed.
+        The :class:`~flash.data.process.Preprocess` hooks covers from data-loading to model forwarding.
+   * - :class:`~flash.data.process.Postprocess`
+     - The :class:`~flash.data.process.Postprocess` provides a simple hook-based API to encapsulate your post-processing logic.
+        The :class:`~flash.data.process.Postprocess` hooks covers from model outputs to predictions export.
+
+*******************************************
+How to use out-of-the-box flashdatamodules
+*******************************************
+
+Flash provides several DataModules with helpers functions.
+Checkout the :ref:`image_classification` section or any other tasks to learn more about them.
+
+********************************
+Why Preprocess and PostProcess ?
+********************************
+
+Currently, it is common practice to implement a `Dataset <https://pytorch.org/docs/stable/data.html#torch.utils.data.Dataset>`_
+and provide them to a `DataLoader <https://pytorch.org/docs/stable/data.html#torch.utils.data.DataLoader>`_.
+
+However, after model training, it requires a lot of engineering overhead to make inference on raw data and deploy the model in production environnement.
+
+The :class:`~flash.data.process.Preprocess` and :class:`~flash.data.process.Postprocess` have been created to resolve these issues.
+By providing a series of hooks that can be overridden with custom data processing logic, the user has much more granular control over their data processing flow.
+
+But it also makes your code more readable, modular and easy to extend.
```

Comment:
Maybe these can be bullet points:

Encapsulate the data processing transforms and the datasets is useful for:

* Making inference on raw data simple
* Make the code more readable, modular and self-contained
* 
anything else?

Reply:
Done !